### DISCUSSION TITLE
question regarding supported Postgresql versions

### DISCUSSION THREAD

**Discussion by swiedernix** [k3s-io/k3s#8343] [General]
According to the documentation :  https://docs.k3s.io/datastore , the supported versions for PostgreSQL as an external database are: 10.7, 11.5, and 14.2

PostgreSQL 14.9 fixes six CVEs since 14.2;
did not check v11 though; 
PostgreSQL v10 is EOL

is that information still valid?

**Comment by brandond** [comment]
That's what our QA folks have most recently validated. You can find the versions that we run automated tests against at:
* https://github.com/k3s-io/k3s/blob/master/scripts/test-setup-sonobuoy-postgres#L19
* https://github.com/k3s-io/kine/blob/master/scripts/test-run-postgres#L26-L31

**Reply by swiedernix** [reply]
Thanks for you reply Brad,

I see some differences betreff VERSION_LIST and the versions from the datastore webpage, they're  not in sync:
webpage  10.7  11.5    14.2
VERSION_LIST="\
    postgres 13.2
    postgres 12.6
    postgres 11.11
    postgres 10.16
    postgres 9.6
    postgres 9.5"

I would tend to remove EOL versions 9/10 and only add the latest minor version from each supported major version.

VERSION_LIST="\
    postgres 16.0
    postgres 15.4
    postgres 14.9
    postgres 13.12
    postgres 12.16
    postgres 11.21

is that OK ?

**Reply by brandond** [reply]
> I see some differences betreff VERSION_LIST and the versions from the datastore webpage, they're not in sync

That's correct. I'm not sure how often our QA team refreshes their database version list - cc @rancher-max 

> I would tend to remove EOL versions 9/10 and only add the latest minor version from each supported major version.

That sounds fine. I'll admit that we don't stay super on top of databases releases; if you wanted to open a PR against k3s-io/kine to bump the database releases that'd be appreciated.

**Comment by rancher-max** [comment]
Yeah, we're due for a refresh. Things will likely work with later versions of DBs, but we only document what we have explicitly validated. https://github.com/k3s-io/k3s/issues/5859 was the last time we did this, so I created https://github.com/k3s-io/k3s/issues/8360 and put it into our backlog to prioritize. Thank you for the awareness!
